config/config.go

```diff
@@ -40,6 +40,7 @@ type DiceDBConfig struct {
 	WALRetentionMode                  string `mapstructure:"wal-retention-mode" default:"num-segments" description:"the new horizon for wal segment post cleanup. values: num-segments, time, checkpoint"`
 	WALMaxSegmentCount                int    `mapstructure:"wal-max-segment-count" default:"10" description:"the maximum number of segments to retain, if the retention mode is 'num-segments'"`
 	WALMaxSegmentRetentionDurationSec int    `mapstructure:"wal-max-segment-retention-duration-sec" default:"600" description:"the maximum duration (in seconds) for wal segments retention"`
+	WALRecovery                       bool   `mapstructure:"wal-recovery" default:"true" description:"enable recovery from wal"`
```

Comment:
We should not have this option. By default if WAL is enabled, recovery should happen. We can get rid of this and reuse EnableWAL to put restoration behind the flag.